ecommerce/frontend/src/components/VendorContent/VendorSplash.js

```diff
@@ -0,0 +1,92 @@
+import { useAppContext } from '../../context/AppContext'
+import React, { useEffect, useState } from 'react'
+import { Spin, Button, Rate, Switch } from 'antd'
+import { api } from '../../api'
+import { useHistory } from 'react-router-dom'
+import { EditOutlined } from '@ant-design/icons'
+import { round } from '../../utils'
+import './VendorSplash.less'
+
+const getVendorRatingLevel = ({ rating }) => {
+    if (rating <= 5.0) {
+        return 'low'
+    }
+    if (rating <= 8.0) {
+        return 'medium'
+    }
+    return 'high'
+}
+
+export const VendorSplash = ({ vendorId, onEditModeChange, editable }) => {
+    const [vendorHeaderDetails, setVendorHeaderDetails] = useState({})
+    const [isLoading, setIsLoading] = useState(true)
+
+    const { user } = useAppContext()
+    const isVendorAndOwner = user.type === 'vendor' && vendorId === user.id.toString()
+    const history = useHistory()
```

Comment:
make sure vendorId is passed as an integer from parent component instead (to avoid subtle bugs where you forget it is a string)

Reply:
done